candle-pyo3/src/lib.rs

```diff
@@ -772,6 +772,39 @@ impl PyTensor {
         Ok(PyTensor(self.0.copy().map_err(wrap_err)?))
     }
 
+    #[pyo3(signature = (*args, **kwargs), text_signature = "(self, *args, **kwargs)")]
+    /// Performs Tensor dtype and/or device conversion.
+    /// &RETURNS&: Tensor
+    fn to(&self, args: &PyTuple, kwargs: Option<&PyDict>) -> PyResult<Self> {
+        let mut t = self.0.clone();
```

Comment:
I wouldn't see a need for this clone and having a mutable variable here but maybe I'm missing something?

Reply:
I probably made it mutable, because the tensor can either be modified by the kwargs, the args or not at all. But i will probably build some sort of parser that ensures, that only one dtype and device are provided.